src/server/game/Tools/CharacterDatabaseCleaner.cpp

```diff
@@ -154,3 +157,11 @@ void CharacterDatabaseCleaner::CleanCharacterQuestStatus()
 {
     CharacterDatabase.DirectExecute("DELETE FROM character_queststatus WHERE status = 0");
 }
+
+void CharacterDatabaseCleaner::CleanCharacterItem()
+{
+    CharacterDatabase.DirectExecute("CREATE TEMPORARY TABLE temp_guid_mapping AS SELECT guid AS old_guid, ROW_NUMBER() OVER (ORDER BY guid) AS new_guid FROM item_instance;");
+    CharacterDatabase.DirectExecute("UPDATE item_instance i JOIN temp_guid_mapping m ON i.guid = m.old_guid SET i.guid = m.new_guid;");
+    CharacterDatabase.DirectExecute("UPDATE character_inventory ci JOIN temp_guid_mapping m ON ci.item = m.old_guid SET ci.item = m.new_guid;");
```

Comment:
We need to do the same for `mail_items`, `guild_bank_item`, `recovery_item`, `auctionhouse` and probably some more tables (use a collective brain?).

Reply:
ah true they would all hava record in item_instance